### PR TITLE
Use our packaged version of `selfregistration`

### PR DESCRIPTION
**Pull request by rosskarchner**
This will be our first usage of the new internal package repository.

## Changes

- git url for 'selfregistration' replaced with a standard package name and version

## Testing

in your virtualenv for developing cfgov-refresh, follow the instructions at [GHE]/CFGOV/platform/wiki/Using-our-Python-package-repository

Then, install the package:

`pip install selfregistration==1.3.1`

You should then be able to run `cfgov/manage.py runserver`, and validate that you can reach the page provided by this app, at http://localhost:8000/company-signup/

## Todos

-

## Checklist

* [ ] PR has an informative and human-readable title
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [ ] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

File: requirements/optional-private.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # git config --global url.https://<private hostname>/.insteadOf https://private.repo/
 
 git+https://private.repo/CFPB/agreement_database.git@2.2.7#egg=agreement-database
-git+https://private.repo/CFPB/cfgov-selfregistration.git@v1.2#egg=selfregistration
+selfregistration==@1.3.1
```
Comment (willbarton) [inline requirements/optional-private.txt]: Is this using it via our CFPyPiB*?

* soliciting suggestions for better naming.

Comment (rosskarchner) [inline requirements/optional-private.txt]: yep! The build is already configured to use the repo, so this will be the first test

Comment (willbarton) [inline requirements/optional-private.txt]: Actually, @rosskarchner, thinking about this a little more: how will this work with `pip install -r` locally? What configuration do we need to add/document for that use-case?

Comment (rosskarchner) [inline requirements/optional-private.txt]: Same as above-- we'll probably need to bake these instructions more fully into our docs:
[GHE]/CFGOV/platform/wiki/Using-our-Python-package-repository

Comment (willbarton) [inline requirements/optional-private.txt]: I do apologize, both of my questions could easily have been answered by reading the PR. 

Context-switching kills brain cells. Or something.

**Pull request merged**